### PR TITLE
Fogging component

### PR DESCRIPTION
**Pull request by nabily-met**
## Goal

A fogging component is added.

If the vapor fraction is to be calibrated, if the water flow is not sufficient, x may be > 1

## Type of change 

- [ ] Bugfix
- [x] New feature
- [ ] Refactoring change
- [ ] Release & Version Update (don't forget to change the version number in `package.mo`)

Will it break anything in previous models ?

- [ ] Breaking change (If yes, make sure to point it out in the changelog)
- [x] Non-Breaking change

## Checklist

- [x] I have added the appropriate tags, reviewers, projects (and detailed the size and priority of my PR) and linked issues to this PR
- [x] I have performed a self-review of my own code
- [x] I have checked that all existing tests pass.
- [x] I have added/updated tests that prove my development works and does not break anything.
- [ ] I have made corresponding changes or additions to the documentation (in [Notion documentation](https://www.notion.so/metroscope/Metroscope-Modeling-Library-Documentation-MML3-WIP-50c8703c294446059d3b4a70d6ae4a71))
- [ ] I have added corresponding entries to the [Changelog](../CHANGELOG.md)
- [x] I have checked for conflicts with target branch, and merged/rebased in consequence

You can also fill these out after creating the PR, but make sure to **check them all before submitting your PR for review**.

### CODE REVIEW

**Review by hadrienp-met — COMMENTED** [review]
See with @pierre-eliep-met for the ` "The input connector ... is not connected from the outside." `errors

File: MetroscopeModelingLibrary/Tests/Multifluid/HeatExchangers/Fogging_reverse.mo [hunk 1]
```diff
@@ -4,17 +4,17 @@ model Fogging_reverse
   // Boundary conditions
   input Real P_fg_in(start = 1.1, min = 0, nominal = 10) "barA";
   input Utilities.Units.MassFlowRate Q_fg_in(start = 660) "kg/s";
-  input Utilities.Units.Temperature T_fg_in(start = 22.21) "degC";
+  input Real T_fg_in(start = 28.8)  "degC";
```
Comment (hadrienp-met) [inline MetroscopeModelingLibrary/Tests/Multifluid/HeatExchangers/Fogging_reverse.mo:7]: I think the unit declaration is useful, it gives warning when you use the wrong unit in the sensor

Comment (pierre-eliep-met) [reply MetroscopeModelingLibrary/Tests/Multifluid/HeatExchangers/Fogging_reverse.mo:7]: Yes but `Utilities.Units.Temperature` is in `K`, not in `degC`

File: MetroscopeModelingLibrary/MultiFluid/HeatExchangers/Fogging.mo [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+        origin={40,44})));
+  FlueGases.Connectors.Outlet C_fg_out annotation (Placement(transformation(extent={{90,-10},{110,10}}), iconTransformation(extent={{90,-10},{110,10}})));
+  WaterSteam.Pipes.PressureCut fogging_nozzle_w annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=270,
+        origin={0,14})));
+  FlueGases.Pipes.PressureCut pressureCut annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=270,
+        origin={40,20})));
```
Comment (pierre-eliep-met) [inline MetroscopeModelingLibrary/MultiFluid/HeatExchangers/Fogging.mo:49]: Add `Q_w_in_0` and `Q_fg_in_0` modifiers in components definition to fully benefit from them

File: MetroscopeModelingLibrary/MultiFluid/HeatExchangers/Fogging.mo [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+  // Definitions
+  h_vap_sat = WaterSteamMedium.dewEnthalpy(WaterSteamMedium.setSat_p(water_evaporation.P_in));
+  h_liq_sat = WaterSteamMedium.bubbleEnthalpy(WaterSteamMedium.setSat_p(water_evaporation.P_in));
+  fogging_nozzle_w.Q = Q_w_in;
+
+  // Parameters
+  x_vapor = (water_evaporation.h_out - h_liq_sat)/(h_vap_sat - h_liq_sat);
+
+  // Energy balance
+  water_evaporation.W = - evaporative_cooling.W;
```
Comment (pierre-eliep-met) [inline MetroscopeModelingLibrary/MultiFluid/HeatExchangers/Fogging.mo:64]: Write it as an energy balance :
```suggestion
  water_evaporation.W + evaporative_cooling.W = 0;
```

**Review by pierre-eliep-met — COMMENTED** [review]
I have not tested the values locally, will try to test it soon

File: MetroscopeModelingLibrary/MultiFluid/HeatExchangers/Fogging.mo [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+model Fogging
+  package FlueGasesMedium =
+      MetroscopeModelingLibrary.Utilities.Media.FlueGasesMedium;
+  package WaterSteamMedium =
+      MetroscopeModelingLibrary.Utilities.Media.WaterSteamMedium;
+  import MetroscopeModelingLibrary.Utilities.Units;
+  import MetroscopeModelingLibrary.Utilities.Units.Inputs;
+
+  // Boundary Conditions
+  Inputs.InputMassFlowRate Q_fg_in(start=Q_fg_in_0) "Inlet fg mass flow rate at the inlet";
```
Comment (pierre-eliep-met) [inline MetroscopeModelingLibrary/MultiFluid/HeatExchangers/Fogging.mo:11]: Sorry I had not realized that : don't ass `_in` to flows, since it is the same at the inlet and outlet. Also the comment is redundant
```suggestion
  Inputs.InputMassFlowRate Q_fg(start=Q_fg_0) "Inlet flue gases mass flow rate";
```
Note that you will have to update it everywhere in the file

File: MetroscopeModelingLibrary/MultiFluid/HeatExchangers/Fogging.mo [hunk 5]
```diff
@@ -0,0 +1,363 @@
...
+
+  // Parameters
+  Inputs.InputMassFraction x_vapor(start=1); // Vapor mass fraction
+
+  // Definitions
+  Units.SpecificEnthalpy h_vap_sat;
+  Units.SpecificEnthalpy h_liq_sat;
+  Units.MassFlowRate Q_w_in(start=Q_w_in_0) "Inlet water mass flow rate at the inlet";
```
Comment (pierre-eliep-met) [inline MetroscopeModelingLibrary/MultiFluid/HeatExchangers/Fogging.mo:19]: Same here
```suggestion
  Units.MassFlowRate Q_w(start=Q_w_0) "Inlet water mass flow rate";
```

File: MetroscopeModelingLibrary/MultiFluid/HeatExchangers/Fogging.mo [hunk 6]
```diff
@@ -0,0 +1,363 @@
...
+
+  // Initialization parameters
+  // Flow Rates
+  parameter Units.MassFlowRate Q_fg_in_0 = 500;
+  parameter Units.MassFlowRate Q_w_in_0 = 1;
```
Comment (pierre-eliep-met) [inline MetroscopeModelingLibrary/MultiFluid/HeatExchangers/Fogging.mo:24]:
```suggestion
  parameter Units.MassFlowRate Q_fg_0 = 500;
  parameter Units.MassFlowRate Q_w_0 = 1;
```

**Review by pierre-eliep-met — APPROVED** [review]
All good, you just have to remove the `_in` from all your flow variables (sorry for not pointing it out at first review) and then you can merge directly.
Did you test the behaviour with your castejon data ?

File: MetroscopeModelingLibrary/MultiFluid/HeatExchangers/Fogging.mo [hunk 7]
```diff
@@ -0,0 +1,363 @@
...
+  // Definitions
+  h_vap_sat = WaterSteamMedium.dewEnthalpy(WaterSteamMedium.setSat_p(water_evaporation.P_in));
+  h_liq_sat = WaterSteamMedium.bubbleEnthalpy(WaterSteamMedium.setSat_p(water_evaporation.P_in));
+  fogging_nozzle_w.Q = Q_w_in;
+
+  // Parameters
+  x_vapor = (water_evaporation.h_out - h_liq_sat)/(h_vap_sat - h_liq_sat);
+
+  // Energy balance
+  water_evaporation.W + evaporative_cooling.W=0;
```
Comment (hadrienp-met) [inline MetroscopeModelingLibrary/MultiFluid/HeatExchangers/Fogging.mo:64]:
```suggestion
  water_evaporation.W + evaporative_cooling.W = 0;
```

File: MetroscopeModelingLibrary/Tests/Multifluid/HeatExchangers/Fogging_reverse.mo [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+  input Real P_water(start = 5, min = 0, nominal = 10) "barA";
+  input Real Q_water(start = 2.5) "kg/s";
+  input Real T_water(start = 19.86, min = 0, nominal = 50) "degC";
+
+  // Calibrated parameters
+  input Utilities.Units.MassFraction x_vapor(start=1);
+
+  // Calibration inputs
+  output Real T_fg_out(start=23.5) "degC";
```
Comment (hadrienp-met) [inline MetroscopeModelingLibrary/Tests/Multifluid/HeatExchangers/Fogging_reverse.mo:17]: Here the comments are not very clear, do we calibrate something or not ? If yes, we should have a clear reverse model and a direct model

**Review by hadrienp-met — CHANGES_REQUESTED** [review]

**Review by hadrienp-met — APPROVED** [review]
Ok, I just added the Fogging_direct in package file

File: MetroscopeModelingLibrary/Tests/Multifluid/HeatExchangers/Fogging_reverse.mo (hunk 8 above)
Comment (hadrienp-met) [reply MetroscopeModelingLibrary/Tests/Multifluid/HeatExchangers/Fogging_reverse.mo:17]: Well in fact it is still not clear for me here, @LucileD-met do we calibrate `x_vapor` ?

File: MetroscopeModelingLibrary/Tests/Multifluid/HeatExchangers/Fogging_direct.mo [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+  input Real P_fg_in(start = 1.1, min = 0, nominal = 10) "barA";
+  input Utilities.Units.MassFlowRate Q_fg_in(start = 660) "kg/s";
+  input Real T_fg_in(start = 28.8)  "degC";
+
+  input Real P_water(start = 5, min = 0, nominal = 10) "barA";
+  input Real Q_water(start = 2.5) "kg/s";
+  input Real T_water(start = 19.86, min = 0, nominal = 50) "degC";
+
+  // Parameters
+  input Utilities.Units.MassFraction x_vapor(start=1);
```
Comment (hadrienp-met) [inline MetroscopeModelingLibrary/Tests/Multifluid/HeatExchangers/Fogging_direct.mo:14]:
```suggestion
  parameter Utilities.Units.MassFraction x_vapor(start=1);
```

**Review by hadrienp-met — APPROVED** [review]

**Comment by hadrienp-met** [comment]
Update
